lib/cpus/aarch64/denver.S

```diff
@@ -189,6 +189,15 @@ func denver_disable_dco
 	ret
 endfunc denver_disable_dco
 
+func check_errata_cve_2017_5715
+#if WORKAROUND_CVE_2017_5715
+	mov	x0, #ERRATA_APPLIES
```

Comment:
It is not enough to assume that erratum applies if WORKAROUND_CVE_2017_5715 is set.

In denver_reset_func you have this:

`
        /*
         * Check if the CPU supports the special instruction
         * required to flush the indirect branch predictor and
         * RSB. Support for this operation can be determined by
         * comparing bits 19:16 of ID_AFR0_EL1 with 0b0001.
         */
        mrs     x0, id_afr0_el1
        mov     x1, #0x10000
        and     x0, x0, x1
        cmp     x0, #0
        adr     x1, workaround_bpflush_runtime_exceptions
        mrs     x2, vbar_el3
        csel    x0, x1, x2, ne
        msr     vbar_el3, x0
`

I think a similar check is required to really be sure that the erratum workaround will apply.

Reply:
just to clarify, ERRATA_APPLIES means CPU is vulnerable and the workaround is active, ERRATA_NOT_APPLIES means CPU is not vulnerable and ERRATA_MISSING means CPU is vulnerable but workaround is not compiled in.

Reply:
@dp-arm 

Does it make sense to keep the check if special instruction only at one place? Not sure why we need the same check at two places

Reply:
@vwadekar yeah you can factor it out and call it from two places